DSCResources/MSFT_xVMSwitch/MSFT_xVMSwitch.psm1

```diff
@@ -228,48 +228,54 @@ function Test-TargetResource
             # If switch should be present, check the switch type
             if($Ensure -eq 'Present')
             {
-                # If the BandwidthReservsationMode is correct, or if $switch.BandwidthReservationMode is $null which means it isn't supported on the OS
-                Write-Verbose -Message "Checking if Switch $Name has correct BandwidthReservationMode ..."
-                if($switch.BandwidthReservationMode -eq $BandwidthReservationMode -or $switch.BandwidthReservationMode -eq $null)
+                ## Only check the BandwidthReservationMode if specified
+                if($PSBoundParameters.ContainsKey('BandwidthReservationMode'))
```

Comment:
@iainbrighton, in that case, I'd remove the default assignment to BandwidthReservationMode, and remove NA as even being an option. Then just throw an error when a configuration specifies it on 2008 R2.